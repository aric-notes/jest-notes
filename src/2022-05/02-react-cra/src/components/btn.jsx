export const Btn = ({ content, ...props }) => (
  <button {...props}>
    hi
  </button>
);