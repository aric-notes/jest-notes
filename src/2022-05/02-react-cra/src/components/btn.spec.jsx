import { render, screen } from '@testing-library/react';
import { Btn } from './btn';

test('btn should have hi text', () => {
  render(<Btn />);
  // screen.debug();
  expect(screen.getByRole('button')).toHaveTextContent('hi');
});

test('btn snapshot test without props', () => {
  const { container } = render(<Btn />);
  expect(container).toMatchSnapshot();
});

test('btn snapshot test with props', () => {
  const { container } = render(<Btn hidden={false} disabled />);
  expect(container).toMatchSnapshot();
});