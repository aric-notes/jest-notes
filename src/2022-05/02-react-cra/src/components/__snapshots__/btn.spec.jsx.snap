// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`btn snapshot test with props 1`] = `
<div>
  <button
    disabled=""
  >
    hi
  </button>
</div>
`;

exports[`btn snapshot test without props 1`] = `
<div>
  <button>
    hi
  </button>
</div>
`;
